packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -20,6 +20,11 @@ export class TreeActions {
   private _currentModel: TreeModel;
   private _reset = new Subject<void>();
 
+  private _nodeLoader: Subject<LoadNodesOptions>;
+  private _parentId?: string;
+  private _timeTracker?: TimeTracker;
+  private _initialRootNode?: TreeModelRootNode;
+
```

Comment:
@grigasp, @saskliutas, @JonasDov could not find a better solution to provide the next/complete functions their props while keeping the `debounceWrapper` generic any suggestions?

Reply:
I think, instead of setting the arguments as class members, you should send them to `_nodeLoader`. That way, the debouncing loader could ensure that only similar requests are debounced (e.g. requests for different parents aren't dropped).

Reply:
Moved inside TreeActions.